sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,509 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import functools
+from typing import Optional, Any
+
+try:
+    from urllib.parse import urlparse, unquote
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+    from urllib2 import unquote  # type: ignore
+
+from azure.core.paging import ItemPaged
+from azure.table._deserialize import _convert_to_entity
+from azure.table._entity import Entity
+from azure.table._generated import AzureTable
+from azure.table._generated.models import AccessPolicy, SignedIdentifier
+from azure.table._message_encoding import NoEncodePolicy, NoDecodePolicy
+from azure.table._serialize import _get_match_headers, _add_entity_properties
+from azure.table._shared.base_client import StorageAccountHostsMixin, parse_query, parse_connection_str
+
+from azure.table._shared.request_handlers import serialize_iso
+from azure.table._shared.response_handlers import process_storage_error
+from azure.core.exceptions import HttpResponseError, ResourceNotFoundError
+from azure.table._version import VERSION
+from azure.core.tracing.decorator import distributed_trace
+
+from ._models import TableEntityPropertiesPaged, UpdateMode
+
+from ._shared.response_handlers import return_headers_and_deserialized
+
+
+class TableClient(StorageAccountHostsMixin):
+    def __init__(
+            self, account_url,  # type: str
+            table_name,  # type: str
+            credential=None,  # type: Optional[Any]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Create TableClient from a Credential.
+
+        :param account_url:
+            A url to an Azure Storage account.
+        :type credential: str
+        :param table_name: The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :type credential: Any
```

Comment:
We can probably narrow the type down a bit. would Union[str, TokenCredential] be accurate?

Reply:
I will check 👍